scala/providers.bzl

```diff
@@ -7,20 +7,26 @@ ScalacProvider = provider(
     ],
 )
 
-def _declare_scalac_provider(ctx):
+DepsInfo = provider(
+    doc = "Defines depset required by rules",
+    fields = {
+        "deps": "Depset",
+        "depset_id": "Identifier by which rules access this depset",
+    },
+)
+
+def _declare_deps_provider(ctx):
     return [
-        ScalacProvider(
-            default_classpath = ctx.attr.default_classpath,
-            default_repl_classpath = ctx.attr.default_repl_classpath,
-            default_macro_classpath = ctx.attr.default_macro_classpath,
+        DepsInfo(
+            deps = ctx.attr.deps,
+            depset_id = ctx.attr.depset_id,
         ),
     ]
 
-declare_scalac_provider = rule(
-    implementation = _declare_scalac_provider,
+declare_deps_provider = rule(
+    implementation = _declare_deps_provider,
     attrs = {
-        "default_classpath": attr.label_list(allow_files = True),
-        "default_repl_classpath": attr.label_list(allow_files = True),
-        "default_macro_classpath": attr.label_list(allow_files = True),
+        "deps": attr.label_list(allow_files = True),
+        "depset_id": attr.string(mandatory = True),
```

Comment:
Unless this is actually somehow a [depset](https://docs.bazel.build/versions/master/skylark/lib/depset.html), I would suggest renaming.

Reply:
Thanks, I will rename it